ui/js/controllers/perf/compare.js

```diff
@@ -116,31 +140,42 @@ perf.controller('CompareChooserCtrl', [
             };
 
             $scope.runCompare = function() {
-                ThResultSetModel.getResultSetsFromRevision($scope.originalProject.name, $scope.originalRevision).then(
+                var orignalRevisionPromise;
+                if($scope.originalRevision.length){
+                    orignalRevisionPromise =  ThResultSetModel.getResultSetsFromRevision($scope.originalProject.name, $scope.originalRevision).then(
                     function(resultSets) {
                         $scope.originalRevisionError = undefined;
                     },
                     function(error) {
                         $scope.originalRevisionError = error;
-                    }
-                );
-
-                ThResultSetModel.getResultSetsFromRevision($scope.newProject.name, $scope.newRevision).then(
+                    });
+                }
+                var newRevisionPromise =  ThResultSetModel.getResultSetsFromRevision($scope.newProject.name, $scope.newRevision).then(
```

Comment:
Again, spacing. There's two spaces here between "=" and "ThResultSetModel", when there should only be one.